src/Store/MongoDB/Store.php

```diff
@@ -135,6 +138,37 @@ public function query(Vector $vector, array $options = []): array
         return $documents;
     }
 
+    /**
+     * @param array{fields?: array<mixed>} $options
+     */
+    public function initialize(array $options = []): void
+    {
+        try {
+            if ([] !== $options && !array_key_exists('fields', $options)) {
+                throw new \InvalidArgumentException('The only supported option is "fields"');
+            }
+
+            $this->getCollection()->createSearchIndex(
```

Comment:
would move the guard close out of the try catch block, since the intention is limited to `createSearchIndex`

```suggestion
        if ([] !== $options && !array_key_exists('fields', $options)) {
            throw new \InvalidArgumentException('The only supported option is "fields"');
        }

        try {
            $this->getCollection()->createSearchIndex(
```